### PR TITLE
feat: special case max timestamp range for table_names and field_columns

### PR DESCRIPTION
**Pull request by tustvold**
Similar in principle to https://github.com/influxdata/influxdb_iox/pull/3485 this special cases max timestamp ranges for table_names and field_columns.

This "fixes" the behaviour of `SHOW MEASUREMENTS` and `SHOW FIELD KEYS` with respect to a row with the maximum timestamp of 9223372036854775806.

This is because despite neither of these queries specifying a time predicate, the RPC request has one that exludes this row.

```
2022-02-04T13:58:03.867065Z  INFO influxdb_iox::influxdb_ioxd::server_type::database::rpc::storage::service: measurement_names db_name=54425547585a6420_000000000153158f range=Some(TimestampRange { start: -9223372036854775806, end: 9223372036854775806 }) predicate=<NONE>
```

Interestingly the actual measurement_fields request has the "correct" time range, but this seems to follow a measurement_names that has the problematic predicate

```
2022-02-04T13:58:03.875796Z  INFO influxdb_iox::influxdb_ioxd::server_type::database::rpc::storage::service: measurement_fields db_name=54425547585a6420_000000000153158f range=Some(TimestampRange { start: -9223372036854775808, end: 9223372036854775807 }) predicate=<NONE>
```

It is unclear if this is actually an upstream bug, so creating as a draft and we can merge this depending on what is decided.

Relates to #3445

### CODE REVIEW

File: query/src/frontend/influxrpc.rs [hunk 1]
```diff
@@ -219,6 +219,9 @@ impl InfluxRpcPlanner {
     {
         debug!(?rpc_predicate, "planning table_names");
 
+        // Special case predicates that span the entire valid timestamp range
+        let rpc_predicate = rpc_predicate.clear_timestamp_if_max_range();
```
Comment (alamb) [inline query/src/frontend/influxrpc.rs:223]: I wonder if we should just always apply this special case logic? But perhaps then we would get discrepancies the other way if `storage` didn't implement it for all entry points? 🤷

**Review by alamb — APPROVED** [review]

File: query/src/frontend/influxrpc.rs (hunk 1 above)
Comment (tustvold) [reply query/src/frontend/influxrpc.rs:223]: I think the inconsistency arises based on whether storage consults the index or the data itself. Tbh I personally would prefer IOx was consistent even if that introduced discrepancies w.r.t storage. In such a case I'd vote for removing this special case altogether...

Comment (alamb) [reply query/src/frontend/influxrpc.rs:223]: 🔜  (after IOx is running the workload). I think we need to hold our noses for the time being in service of TSM compatiblity

**Comment by alamb** [comment]
Shall we merge this one @tustvold ?

**Comment by tustvold** [comment]
I've re-run this branch against the InfluxRPC test suite and it looks good - so I think let's go for it

**Pull request merged**